pkg/controller/versions/strategy_v2_3.go

```diff
@@ -224,6 +273,71 @@ func (v *versionStrategyV2_3) ValidateUpgrade(ctx context.Context, cl client.Cli
 	return nil
 }
 
+func (v *versionStrategyV2_3) ValidateUpdate(ctx context.Context, cl client.Client, oldSMCP, newSMCP metav1.Object) error {
+	oldClusterScoped, err := isClusterScoped(oldSMCP)
+	if err != nil {
+		return err
+	}
+	newClusterScoped, err := isClusterScoped(newSMCP)
+	if err != nil {
+		return err
+	}
+	if oldClusterScoped != newClusterScoped {
+		return fmt.Errorf("field spec.techPreview.%s is immutable", v2.ControlPlaneModeKey)
+	}
+	return nil
+}
+
+func isClusterScoped(smcp metav1.Object) (bool, error) {
+	switch s := smcp.(type) {
+	case *v1.ServiceMeshControlPlane:
+		return false, nil
+	case *v2.ServiceMeshControlPlane:
+		return s.Spec.IsClusterScoped()
+	default:
+		return false, fmt.Errorf("unknown ServiceMeshControlPlane type: %T", smcp)
+	}
+}
+
+func (v *versionStrategyV2_3) ValidateRequest(ctx context.Context, cl client.Client, req admission.Request, smcp metav1.Object) admission.Response {
+	clusterScoped, err := isClusterScoped(smcp)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	if clusterScoped {
+		isClusterAdmin, err := v.isRequesterClusterAdmin(ctx, cl, req)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		if !isClusterAdmin {
+			return admission.ValidationResponse(false, "a cluster-scoped SMCP may only be created by users with cluster-admin permissions")
+		}
+	}
+
+	return admission.ValidationResponse(true, "")
+}
+
+func (v *versionStrategyV2_3) isRequesterClusterAdmin(ctx context.Context, cl client.Client, req admission.Request) (bool, error) {
+	sar := &authorizationv1.SubjectAccessReview{
+		Spec: authorizationv1.SubjectAccessReviewSpec{
+			User:   req.AdmissionRequest.UserInfo.Username,
+			UID:    req.AdmissionRequest.UserInfo.UID,
+			Extra:  common.ConvertUserInfoExtra(req.AdmissionRequest.UserInfo.Extra),
+			Groups: req.AdmissionRequest.UserInfo.Groups,
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Verb:     "*",
+				Group:    "*",
+				Resource: "*",
+			},
+		},
+	}
+	err := cl.Create(ctx, sar)
+	if err != nil {
+		return false, err
+	}
+	return sar.Status.Allowed && !sar.Status.Denied, nil
```

Comment:
Is the status field guaranteed to be filled when Create() returns?

Reply:
Yes.